gui/include/gui/gl_tex_cache.h

```diff
@@ -105,10 +105,51 @@ class glTexTile {
 
 #define MAX_TEX_LEVEL 10
 
+#ifdef _MSC_VER
+#ifdef new
+#define OCPN_PUSHED_DEBUG_NEW
+#pragma push_macro("new")
+#undef new
+#endif
+#endif
+
 class glTexFactory {
 public:
   glTexFactory(ChartBase *chart, int raster_format);
-  ~glTexFactory();
+
+  // Make class uncopyable
+  glTexFactory(const glTexFactory &) = delete;
+  glTexFactory &operator=(const glTexFactory &) = delete;
+
+  // Reference counting methods used by job manager
```

Comment:
Wouldn't it be better to just use std::shared_ptr instead of implementing our own reference counting?

Reply:
It's not quite compatible with shared_ptr type based on how it is used. I did try to shoe-horn shared_ptr to work but always hit a corner case where it did not suffice.

Reply:
Ok, makes sense.